test/hotspot/jtreg/vmTestbase/nsk/jvmti/GetCurrentContendedMonitor/contmon002.java

```diff
@@ -42,21 +41,32 @@
         }
     }
 
-    public static Wicket startingBarrier;
+    public static boolean startingBarrier = true;
 
     public static void main(String[] args) {
         args = nsk.share.jvmti.JVMTITest.commonInit(args);
 
         System.exit(run(args, System.out) + 95/*STATUS_TEMP*/);
     }
 
+    public static void doSleep() {
+        try {
+            Thread.sleep(10);
+        } catch (Exception e) {
+            throw new Error("Unexpected " + e);
+        }
+    }
+
     public static int run(String argv[], PrintStream ref) {
+        doSleep(); // If it would do any class loading, do it now.
```

Comment:
I think now the spawned thread should not try to resolve any new methods after setting the boolean so we shouldn't have the same ObjectLocker issue.

Reply:
Perhaps:
// If we need to load any classes do execute doSleep(), do it now.

Reply:
Fixed